toolkit/tools/internal/retry/retry_test.go

```diff
@@ -0,0 +1,133 @@
+package retry
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var defaultTestTime = time.Millisecond * 100
+
+// Tests won't run with perfect timing, so we need to allow for some fudge factor when we check maximum duration.
+var timingFudgeFactor = time.Millisecond * 100
+
+func TestZeroBase(t *testing.T) {
+	// Two failures, 0.0 multiplier, should be 0 second delay
+	base := 0.0
+	fails := 2
+	val := calculateExpDelay(fails, defaultTestTime, base)
+	assert.Equal(t, time.Duration(0), val)
+}
+
+func TestZeroI(t *testing.T) {
```

Comment:
Could we make the test names more descriptive? Names like `ZeroBase` or `NegativeI` don't really say much.